utils/message-generator/src/executor.rs

```diff
@@ -203,558 +203,600 @@ impl Executor {
                 // Can't do anything else after the connection drops.
                 if *result == ActionResult::CloseConnection {
                     info!(
-                        "Waiting 1 sec to make sure that remote have time to close the connection"
+                        "Waiting 1 sec to make sure that remote has time to close the connection"
                     );
                     tokio::time::sleep(std::time::Duration::from_millis(1000)).await;
                     recv.recv()
                         .await
                         .expect_err("Expecting the connection to be closed: wasn't");
                     success = true;
                     break;
-                }
 
-                let message = match recv.recv().await {
-                    Ok(message) => message,
-                    Err(_) => {
+                // Conversely, if the connection should sustain but the channel is closed,
+                // We also break since the test has failed
+                } else if *result == ActionResult::SustainConnection {
```

Comment:
Why not have this and `CloseConnection` as part of the match in L241?

Reply:
I see your point!

I initially avoided this approach for the same reason I moved all those lines inside the `else` case

the original scenario that motivated this PR is one where the role being tested:
- receives a message
- does not close the connection (it was closing, which is the bug we're aiming to fix)
- does not send any message back in response

if we had the logic from this `else if` moved down to the `match` in L241, while unconditionally running `let message = match recv.recv().await`, the test would end up hanging on the scenario described above.

but now your suggestion inspired me to explore an alternative approach

I committed it on a different branch: https://github.com/plebhash/stratum/commit/e0164a8bc8c99cccc902063db57eab4770b385d5

there, I replicated the `recv.recv().await` on the different `match` arms where a response message is expected, and avoided it on those where it wouldn't necessarily happen, which also avoids the hanging situation

it does feel a bit cleaner, although with the tradeoff of some code repetition

if you prefer, I could bring this alternative approach over to this PR

Reply:
new commit looks great, lets use it

Reply:
done